shacl12-rules/index.html

```diff
@@ -867,942 +951,1014 @@ <h2>Shapes Rules Language Grammar</h2>
+                <td><code>[30]&nbsp;&nbsp;</code></td>
                 <td><code><span class="doc-ref" id="rConstraint">Constraint</span></code></td>
                 <td>&nbsp;&nbsp;::=&nbsp;&nbsp;</td>
                 <td><code class="gRuleBody"><a href="#rBrackettedExpression">BrackettedExpression</a> | <a href="#rBuiltInCall">BuiltInCall</a> | <a href="#rFunctionCall">FunctionCall</a></code></td>
               </tr>
 
               <tr style="vertical-align: baseline">
-                <td><code>[30]&nbsp;&nbsp;</code></td>
+                <td><code>[31]&nbsp;&nbsp;</code></td>
                 <td><code><span class="doc-ref" id="rFunctionCall">FunctionCall</span></code></td>
                 <td>&nbsp;&nbsp;::=&nbsp;&nbsp;</td>
                 <td><code class="gRuleBody"><a href="#riri">iri</a> <a href="#rArgList">ArgList</a></code></td>
               </tr>
 
               <tr style="vertical-align: baseline">
-                <td><code>[31]&nbsp;&nbsp;</code></td>
+                <td><code>[32]&nbsp;&nbsp;</code></td>
                 <td><code><span class="doc-ref" id="rArgList">ArgList</span></code></td>
                 <td>&nbsp;&nbsp;::=&nbsp;&nbsp;</td>
                 <td><code class="gRuleBody"><a href="#rNIL">NIL</a> | <span class="token">'('</span> <a href="#rExpression">Expression</a> ( <span class="token">','</span> <a href="#rExpression">Expression</a> )* <span class="token">')'</span> </code></td>
               </tr>
 
               <tr style="vertical-align: baseline">
-                <td><code>[32]&nbsp;&nbsp;</code></td>
+                <td><code>[33]&nbsp;&nbsp;</code></td>
                 <td><code><span class="doc-ref" id="rExpressionList">ExpressionList</span></code></td>
                 <td>&nbsp;&nbsp;::=&nbsp;&nbsp;</td>
                 <td><code class="gRuleBody"><a href="#rNIL">NIL</a> | <span class="token">'('</span> <a href="#rExpression">Expression</a> ( <span class="token">','</span> <a href="#rExpression">Expression</a> )* <span class="token">')'</span> </code></td>
               </tr>
 
               <tr style="vertical-align: baseline">
-                <td><code>[33]&nbsp;&nbsp;</code></td>
+                <td><code>[34]&nbsp;&nbsp;</code></td>
                 <td><code><span class="doc-ref" id="rTriplesSameSubject">TriplesSameSubject</span></code></td>
                 <td>&nbsp;&nbsp;::=&nbsp;&nbsp;</td>
-                <td><code class="gRuleBody"><a href="#rVarOrTerm">VarOrTerm</a> <a href="#rPropertyListNotEmpty">PropertyListNotEmpty</a> | <a href="#rTriplesNode">TriplesNode</a> <a href="#rPropertyList">PropertyList</a> | <a href="#rReifiedTripleBlock">ReifiedTripleBlock</a></code></td>
+                <td><code class="gRuleBody"><a href="#rVarOrTerm">VarOrTerm</a> <a href="#rPropertyListNotEmpty">PropertyListNotEmpty</a> |	<a href="#rTriplesNode">TriplesNode</a> <a href="#rPropertyList">PropertyList</a> |	<a href="#rReifiedTripleBlock">ReifiedTripleBlock</a></code></td>
```

Comment:
So I guess with the `RDF Rules` serialization of `SHACL Rules`, we would take care of producing standard (not generalized!) RDF, right?

Reply:
Right, in both the RDF syntax is not symmetric RDF (or generalized RDF) and the output inferred graph is a an RDF graph, so RDF triples only.

The RDF syntax does not have triple patterns in it, only written out as `[ :subject S ; :predciate P ; :object O ]`.

For inferring, same as `CONSTRUCT` - non-triples get dropped.

A consequence is that writing a rule with `"123" rdf:type xsd:integer` does not match teh base or the infer graph. The latter may be a problem.